third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -88,6 +88,24 @@ impl Debug for LoadedFunctionOwner {
     }
 }
 
+impl LoadedFunctionOwner {
+    #[inline(always)]
+    pub fn module_or_script_id(&self) -> &ModuleId {
+        match self {
+            LoadedFunctionOwner::Module(m) => Module::self_id(m),
+            LoadedFunctionOwner::Script(_) => language_storage::pseudo_script_module_id(),
+        }
+    }
+
+    #[inline(always)]
+    pub fn module_or_script_id_hash(&self) -> [u8; 32] {
+        match self {
+            LoadedFunctionOwner::Module(m) => m.id_hash,
+            LoadedFunctionOwner::Script(_) => [0u8; 32],
```

Comment:
We should hash `pseudo_script_module_id`? Otherwise hash(id) != this hash and we can have a problem when we do re-entrancy checks?

Reply:
I was doubting about the same, but felt like what I'm doing right now may still be safe, as collisions are practically impossible.

> Otherwise hash(id) != this hash and we can have a problem when we do re-entrancy checks?

We compare only the hashes, but you are right it's not good to have `hash(id) != this hash`. Should probably just compute the real hash for consistency.